File: backend/src/main/java/com/platform/investor/service/InvestorService.java
package com.platform.investor.service;

import com.platform.investor.model.Investor;
import com.platform.investor.model.InvestorDTO;
import com.platform.investor.model.InvestorStatus;
import com.platform.investor.repository.InvestorRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import java.util.Optional;
import java.util.UUID;

@Service
public class InvestorService {
    
    @Autowired
    private InvestorRepository investorRepository;
    
    /**
     * Get investor by ID
     */
    public Optional<Investor> getInvestorById(UUID id) {
        return investorRepository.findById(id);
    }
    
    /**
     * Get investor by user ID
     */
    public Optional<Investor> getInvestorByUserId(UUID userId) {
        return investorRepository.findByUserId(userId);
    }
    
    /**
     * Create a new investor profile
     */
    public Investor createInvestor(InvestorDTO investorDTO) {
        Investor investor = new Investor();
        investor.setUserId(investorDTO.getUserId());
        investor.setInvestmentBudget(investorDTO.getInvestmentBudget());
        investor.setInvestmentStage(investorDTO.getInvestmentStage());
        investor.setSectorsInterested(investorDTO.getSectorsInterested());
        investor.setMinTicketSize(investorDTO.getMinTicketSize());
        investor.setMaxTicketSize(investorDTO.getMaxTicketSize());
        investor.setStatus(investorDTO.getStatus() != null ? investorDTO.getStatus() : InvestorStatus.ACTIVE);
        
        return investorRepository.save(investor);
    }
    
    /**
     * Update existing investor
     */
    public Investor updateInvestor(UUID id, InvestorDTO investorDTO) {
        Optional<Investor> existingInvestor = investorRepository.findById(id);
        if (existingInvestor.isPresent()) {
            Investor investor = existingInvestor.get();
            if (investorDTO.getInvestmentBudget() != null) {
                investor.setInvestmentBudget(investorDTO.getInvestmentBudget());
            }
            if (investorDTO.getInvestmentStage() != null) {
                investor.setInvestmentStage(investorDTO.getInvestmentStage());
            }
            if (investorDTO.getSectorsInterested() != null) {
                investor.setSectorsInterested(investorDTO.getSectorsInterested());
            }
            if (investorDTO.getMinTicketSize() != null) {
                investor.setMinTicketSize(investorDTO.getMinTicketSize());
            }
            if (investorDTO.getMaxTicketSize() != null) {
                investor.setMaxTicketSize(investorDTO.getMaxTicketSize());
            }
            if (investorDTO.getPortfolioCompanies() != null) {
                investor.setPortfolioCompanies(investorDTO.getPortfolioCompanies());
            }
            if (investorDTO.getStatus() != null) {
                investor.setStatus(investorDTO.getStatus());
            }
            return investorRepository.save(investor);
        }
        return null;
    }
    
    /**
     * Delete investor
     */
    public boolean deleteInvestor(UUID id) {
        if (investorRepository.existsById(id)) {
            investorRepository.deleteById(id);
            return true;
        }
        return false;
    }
    
    /**
     * Get investors by status
     */
    public Page<Investor> getInvestorsByStatus(InvestorStatus status, Pageable pageable) {
        return investorRepository.findByStatus(status, pageable);
    }
    
    /**
     * Get all investors with pagination
     */
    public Page<Investor> getAllInvestors(Pageable pageable) {
        return investorRepository.findAll(pageable);
    }
}

File: frontend/src/services/api/investorService.ts
import apiClient from './apiClient';
import { Investor, InvestorStatus, PaginatedResponse } from '../../types';

export const investorService = {
  getAllInvestors: async (page: number = 0, size: number = 20): Promise<PaginatedResponse<Investor>> => {
    const response = await apiClient.get('/investors', { params: { page, size } });
    return response.data;
  },
  getInvestorById: async (id: string): Promise<Investor> => {
    const response = await apiClient.get(`/investors/${id}`);
    return response.data;
  },
  getInvestorByUserId: async (userId: string): Promise<Investor> => {
    const response = await apiClient.get(`/investors/user/${userId}`);
    return response.data;
  },
  getInvestorsByStatus: async (status: InvestorStatus, page: number = 0, size: number = 20): Promise<PaginatedResponse<Investor>> => {
    const response = await apiClient.get(`/investors/status/${status}`, { params: { page, size } });
    return response.data;
  },
  createInvestor: async (investor: Partial<Investor>): Promise<Investor> => {
    const response = await apiClient.post('/investors', investor);
    return response.data;
  },
  updateInvestor: async (id: string, investor: Partial<Investor>): Promise<Investor> => {
    const response = await apiClient.put(`/investors/${id}`, investor);
    return response.data;
  },
  deleteInvestor: async (id: string): Promise<void> => {
    await apiClient.delete(`/investors/${id}`);
  },
};
